XRPL.Core.Domain/Responses/AmmInfoResponse.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System.Runtime.Serialization;
+using XRPL.Core.Domain.Models;
+
+namespace XRPL.Core.Domain.Responses
+{
+    /// <summary>
+    /// Represents a response to an Amm info request.
+    /// </summary>
+    public class AmmInfoResponse : ResponseBase<AmmInfoResult>
+    {
+    }
+
+    /// <summary>
+    /// Represents a result of an <see cref="AmmInfoResponse"/> object.
+    /// </summary>
+    [DataContract]
+    public abstract class AmmInfoResult : ResultBase
+    {
+        ///<summary>
+        ///An AMM Description Object for the requested asset pair.
+        ///</summary>
+        [DataMember(Name = "amm")]
+        public AmmDescriptionObject? Amm {  get; set; }
+
+        ///<summary>
+        ///The ledger index of the current in-progress ledger,
+        ///which was used when retrieving this information.
+        ///</summary>
+        [DataMember(Name = "ledger_current_index")]
+        public uint LedgerCurrentIndex { get; set; }
+
+        ///<summary>
+        ///The identifying hash of the ledger version that was used when retrieving this data.
+        ///</summary>
+        [DataMember(Name = "ledger_hash")]
+        public string? LedgerHash { get; set; }
+
+        ///<summary>
+        ///The ledger index of the ledger version used when retrieving this information.
+        ///</summary>
+        [DataMember(Name = "ledger_index")]
+        public uint LedgerIndex { get; set;}
+
+        ///<summary>
+        ///If true, the ledger used for this request is validated and these results are final;
+        ///if omitted or set to false, the data is pending and may change.
+        ///</summary>
+        [DataMember(Name = "validated")]
+        public bool Validated { get; set; }
+    }
+
+
+    ///<summary>
+    ///The amm field is an object describing the current status of an Automated Market Maker (AMM) in the ledger
+    ///</summary>
+    [DataContract]
+    public class AmmDescriptionObject
```

Comment:
This class should contain two types of CurrencyAmounts: XRP/fungible token. This implies a generic class with two generic parameters for the currency types